TWTValidation/JSON Validator/AST Nodes/TWTJSONSchemaArrayASTNode.m

```diff
@@ -54,4 +54,44 @@ - (NSSet *)validTypes
     return [NSSet setWithObject:TWTJSONSchemaTypeKeywordArray];
 }
 
+
+- (NSArray *)childrenReferenceNodes
+{
+    NSMutableArray *nodes = [[super childrenReferenceNodes] mutableCopy];
+    if (self.itemSchema) {
+        [nodes addObjectsFromArray:self.itemSchema.childrenReferenceNodes];
+    } else {
+        [nodes addObjectsFromArray:[self childrenReferenceNodesFromNodeArray:self.indexedItemSchemas]];
+    }
+
+    [nodes addObjectsFromArray:self.additionalItemsNode.childrenReferenceNodes];
+
+    return nodes;
+}
+
+
+- (TWTJSONSchemaASTNode *)nodeForPathComponents:(NSArray *)path
+{
+    TWTJSONSchemaASTNode *node = [super nodeForPathComponents:path];
+    if (node) {
+        return node;
+    }
+
+    NSString *key = path.firstObject;
+    NSArray *remainingPath = [self remainingPathFromPath:path];
+
+    if ([key isEqualToString:TWTJSONSchemaKeywordItems]) {
+        if (self.itemSchema) {
+            return [self.itemSchema nodeForPathComponents:remainingPath];
+        } else {
+            return [self nodeForPathComponents:remainingPath fromNodeArray:self.indexedItemSchemas];
+        }
+    }
+    if ([key isEqualToString:TWTJSONSchemaKeywordAdditionalItems]) {
```

Comment:
Maybe use else-if? I know that you always return from the previous if statement, but it’s not immediately clear given the if/else inside the if.